### DISCUSSION TITLE
'reorder' option must allow sort by position desc

### DISCUSSION THREAD

**Discussion by ammma** [area17/twill#1055] [Ideas]
Presently if you enable `reorder` option in controller then any values put in `$defaultOrders` will be ignored and list will be sorted by position ascending. This cuts use cases this feature can handle by half. 
For example: i need to build an admin page which reflects front-end and to do so i need an ability to reorder items AND to have newly created items to go to the top of the list, not to the bottom.

And most elegant way to do this i think it to make `ModuleController` to work with  `$defaultOrders = ['position'=>'desc']`

**Comment by pboivin** [comment]
Hey @ammma,

Not sure about the most elegant way to tackle this but I believe this could be a quickfix for your use case.

**In your module repository :**
```
    public function afterSave($object, $fields)
    {
        parent::afterSave($object, $fields);

        if ($object->wasRecentlyCreated) {
            $object->position = 0;
            $object->save();

            $ids = $object::orderBy('position')->pluck('id')->toArray();
            $this->setNewOrder($ids);
        }
    }
```

This will ensure that your new items are added to the top of the list, at the cost of a few extra queries.

Hope this can help!

**Reply by pboivin** [reply]
EDIT: Added a check on `$object->wasRecentlyCreated`, as this is obviously not desirable on every save action :)

**Reply by ammma** [reply]
Thanks! i'll try this, for now i'm not sure if Twill will properly handle several records with position equals zero, need to try(because user can just create many records one after another).
But i still believe Twill should allow this as an option, without hacking(should i request a feature?).
Meanwhile i've created custom controller extending `ModuleController` and overrided following functions:

`    /**
     * @return \Illuminate\Http\JsonResponse
     */
    public function reorder()
    {
        if (($values = $this->request->get('ids')) && !empty($values)) {
            if (!empty($this->defaultOrders['position']) && $this->defaultOrders['position'] == 'desc') {
                $values = array_reverse($values);
            }
            $this->repository->setNewOrder($values);
            $this->fireEvent();
            return $this->respondWithSuccess(twillTrans('twill::lang.listing.reorder.success', ['modelTitle' => $this->modelTitle]));
        }

        return $this->respondWithError(twillTrans('twill::lang.listing.reorder.error', ['modelTitle' => $this->modelTitle]));
    }`


`    /**
     * @return array
     */
    protected function orderScope()
    {
        $orders = [];
        if ($this->request->has('sortKey') && $this->request->has('sortDir')) {
            if (($key = $this->request->get('sortKey')) == 'name') {
                $sortKey = $this->titleColumnKey;
            } elseif (!empty($key)) {
                $sortKey = $key;
            }

            if (isset($sortKey)) {
                $orders[$this->indexColumns[$sortKey]['sortKey'] ?? $sortKey] = $this->request->get('sortDir');
            }
        }

        // don't apply default orders if reorder is enabled
        $reorder = $this->getIndexOption('reorder');
        $defaultOrders = ($reorder && empty($this->defaultOrders['position']) ? [] : ($this->defaultOrders ?? []));

        return $orders + $defaultOrders;
    }`


p.s. sorry, can't insert whole code properly
